pkg/controllers/vdb/status_reconciler.go

```diff
@@ -182,6 +183,24 @@ func (s *StatusReconciler) calculateClusterStatus(stat *vapi.VerticaDBStatus) {
 	}
 }
 
+// getSubclusterStatusType returns the subcluster status type depends on its type in subclusters and sandboxes
+func (s *StatusReconciler) getSubclusterStatusType(podName types.NamespacedName) string {
```

Comment:
We want to rely on the sandboxes status to make this decision, instead of podfacts.